Documentation/CRDs/ceph-cluster-crd.md

```diff
@@ -1334,16 +1334,54 @@ In order to configure an external Ceph cluster with Rook, we need to extract som
+- `--restricted-auth-permission`: (optional) Restrict cephCSIKeyrings auth permissions to specific pools, and cluster. Mandatory flags that need to be set are `--rbd-data-pool-name`, and `--cluster-name`. `--cephfs-filesystem-name` flag can also be passed in case of CephFS user restriction, so it can restrict users to particular CephFS filesystem.
+Note: Restricting the csi-users per pool, and per cluster will require creating new csi-users and new secrets for that csi-users, so it can only be used for new deployments.
+
+  Sample run:
     ```console
-    python3 create-external-cluster-resources.py --rbd-data-pool-name <pool_name> --cephfs-filesystem-name <filesystem-name> --rgw-endpoint  <rgw-endpoint> --namespace <namespace> --rgw-pool-prefix <rgw-pool-prefix> --format bash
+    python3 create-external-cluster-resources.py --cephfs-filesystem-name <filesystem-name> --rbd-data-pool-name <pool_name> --cluster-name <cluster-name> --restricted-auth-permission true --format <bash> --rgw-endpoint <rgw_endpoin> --namespace <rook-ceph-external>
     ```
 
-    * `--namespace`: Namespace where CephCluster will run in the consumer cluster, for example `rook-ceph-external`
-    * `--format bash`: The format of the output
-    * `--rbd-data-pool-name`: The name of the RBD data pool
-    * `--cephfs-filesystem-name`: (optional) The name of the filesystem
-    * `--rgw-endpoint`: (optional) The RADOS Gateway endpoint in the format `<IP>:<PORT>`
-    * `--rgw-pool-prefix`: (optional) The prefix of the RGW pools. If not specified, the default prefix is `default`.
+- `--upgrade`: (optional) Upgrades the 'Ceph CSI keyrings (For example: client.csi-cephfs-provisioner) with new permissions needed for the new cluster version and older permission will still be applied.
+
+   Sample run:
+     ```console
+      python3 /etc/ceph/create-external-cluster-resources.py --upgrade
+    ```
+  This will upgrade all the default csi users(non-restricted)
+
+  Restricted users created using --restricted-auth-permission flag (For example: client.csi-cephfs-provisioner-openshift-storage) need to pass mandatory flags: '--rbd-data-pool-name(if it is a rbd user), --cluster-name and --run-as-user' flags while upgrading, in case of cephfs users if you have passed --cephfs-filesystem-name flag while creating csi-user then while upgrading it will be mandatory too
+
+  Sample run:
+  ```console
+    python3 /etc/ceph/create-external-cluster-resources.py --upgrade --rbd-data-pool-name replicapool --cluster-name rookstorage --run-as-user client.csi-rbd-node-rookStorage-replicapool
+  ```
+
+  > **Note**: An existing non-restricted user cannot be downgraded to a restricted user by upgrading. Admin needs to create a new restricted user for this by re-running the script.
+The upgrade flag should only be used to append new permissions to users. It shouldn't be used for changing a CSI-user already applied permissions. For example, you shouldn't change the pool(s) a user has access to.
```

Comment:
Newer docs are using a different format for callouts. It looks like below (note the 4-space indentation on the `!` text to keep it aligned with the bullet, and the 8-space indentation for the body under `!!! note`). It was introduced here by @galexrt : https://github.com/rook/rook/pull/10265

```
    !!! note
        This is a note
```

Reply:
There are more places in other docs too where this is not updated...
Maybe we can open an issue for it.